---
uint32 EntryX
uint32 EntryY
uint32 ExitX
uint32 ExitY
